### DISCUSSION TITLE
Slow cut for frontend-sdl2?

### DISCUSSION THREAD

**Discussion by emh789** [projectM-visualizer/projectm#663] [Q&A]
Hi. I am having a good experience with the SDL2 frontend on Linux.

Is there an option to have it do slow cuts of a specifiable duration between presets?

I had previously seen this option with a much older version of the qt frontend, but I'm not sure if I'm overlooking it now.

Thanks!

**Comment by kblaschke** [comment]
Sure, it's just not in the default properties file. If the smooth transition is not broken in current libprojectM (which is a possibility given the current work we're doing on the core), you can add and change any of these values, defaults are noted:

```properties
# Preset display time in seconds
projectM.displayDuration = 30
# Slow cut duration in seconds
projectM.transitionDuration = 3
# True to enable beat-driven immediate cuts
projectM.hardCutsEnabled = false
# Number of seconds after which a hard cut can happen
projectM.hardCutDuration = 20
# Beat "power" sensitivity for hard cuts (not to be confused with beat sensitivity)
projectM.hardCutSensitivity = 1.0
# Beat detection sensitivity for preset effects
projectM.beatSensitivity = 1.0
# True to shuffle the playlost, false to play presets in order
projectM.shuffleEnabled = true
```

**Reply by emh789** [reply]
Thanks! Can I put a file in `~/.config` (or elsewhere) to provide custom values, or would the current solution be to modify `frontend-sdl2/src/resources/projectMSDL.properties.in` and rebuild the frontend?

**Reply by kblaschke** [reply]
Sure, you can create a file `$XDG_CONFIG_HOME/projectM/projectMSDL.properties`[^xdg] and then put your settings in there. Anything put into this file will override built-in settings or the default properties file (in the executable path). Command line arguments will override any configuration file though.

The order of settings, with highest priority first which means overriding those below if set:
- Command line options,
- User configuration file (projectMSDL.properties), as noted above,
- Default configuration file (projectMSDL.properties) at the executable's location,
- Built-in defaults (code).

[^xdg]: `XDG_CONFIG_HOME` should default to `~/.config`.

**Reply by emh789** [reply]
Thanks!

**Comment by kblaschke** [comment]
Just a heads-up here: The latest development sources now have new transition code, which properly blends presets using (ShaderToy-compatible) shaders. We're close to releasing version 4.1 of the library, and the new frontend UI is also making progress.
